apiserver/facades/client/sshclient/facade_test.go

```diff
@@ -504,15 +504,186 @@ func (s *facadeSuite) TestGetVirtualHostnameForEntity(c *gc.C) {
+	})
+
+	// Test machine target hits correct methods.
+	gomock.InOrder(
+		backend.EXPECT().MachineVirtualPublicHostKeyPEM("1").Return("machine-host-key", nil).Times(1),
+		backend.EXPECT().SSHServerHostKey().Return("server-host-key", nil).Times(1),
+	)
+	res = facade.PublicHostKeyForTarget(params.SSHHostKeyRequestArg{
+		Hostname: "1.8419cd78-4993-4c3a-928e-c646226beeee.juju.local",
+	})
+	c.Assert(res, gc.Equals, params.SSHHostKeyResult{
+		Error:             nil,
+		HostKey:           "machine-host-key",
+		JumpServerHostKey: "server-host-key",
+	})
+}
+
+func (s *facadeSuite) TestHostKeyForTargetUnhappyPaths(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+	backend := mocks.NewMockBackend(ctrl)
+	authorizer := mocks.NewMockAuthorizer(ctrl)
+	broker := mocks.NewMockBroker(ctrl)
+
+	authorizer.EXPECT().AuthClient().Return(true).Times(1)
+
+	facade, err := sshclient.InternalFacade(backend, nil, authorizer, s.callContext,
+		func(context.Context, environs.OpenParams) (sshclient.Broker, error) {
+			return broker, nil
+		},
+	)
+	c.Assert(err, gc.IsNil)
+
+	// Set superuser for all calls.
+	controllerTag := names.NewControllerTag("controller00-0000-0000-000000000000")
+	backend.EXPECT().ControllerTag().Return(controllerTag).AnyTimes()
+	authorizer.EXPECT().HasPermission(permission.SuperuserAccess, controllerTag).AnyTimes()
+
+	// Test failure to parse hostname.
+	res := facade.PublicHostKeyForTarget(params.SSHHostKeyRequestArg{
+		Hostname: "BLAAAH 1.8419cd78-4993-4c3a-928e-c646226beeee.juju.local",
+	})
+	c.Assert(res.Error, gc.ErrorMatches, "failed to parse hostname: could not parse hostname")
+
+	// Test MachineVirtualHostKey fail.
+	gomock.InOrder(
+		backend.EXPECT().MachineVirtualPublicHostKeyPEM("1").Return("", errors.New("an-error")).Times(1),
+	)
+	res = facade.PublicHostKeyForTarget(params.SSHHostKeyRequestArg{
+		Hostname: "1.8419cd78-4993-4c3a-928e-c646226beeee.juju.local",
+	})
+	c.Assert(res.Error, gc.ErrorMatches, "failed to get machine host key: an-error")
+
+	// Test UnitVirtualHostKey fail.
+	gomock.InOrder(
+		backend.EXPECT().UnitVirtualPublicHostKeyPEM("openfga/1").Return("", errors.New("an-error")).Times(1),
+	)
+	res = facade.PublicHostKeyForTarget(params.SSHHostKeyRequestArg{
+		Hostname: "1.openfga.8419cd78-4993-4c3a-928e-c646226beeee.juju.local",
+	})
+	c.Assert(res.Error, gc.ErrorMatches, "failed to get unit host key: an-error")
+
+	// Test SSHServerHostKey fail.
+	gomock.InOrder(
+		backend.EXPECT().UnitVirtualPublicHostKeyPEM("postgresql/1").Return("postgres-container-host-key", nil).Times(1),
+		backend.EXPECT().SSHServerHostKey().Return("", errors.New("an-error")).Times(1),
+	)
+	res = facade.PublicHostKeyForTarget(params.SSHHostKeyRequestArg{
+		Hostname: "charm.1.postgresql.8419cd78-4993-4c3a-928e-c646226beeee.juju.local",
+	})
+	c.Assert(res.Error, gc.ErrorMatches, "failed to get controller jumpserver host key: an-error")
+}
+
+func (s *facadeSuite) TestHostKeyForTargetUserAuth(c *gc.C) {
```

Comment:
Even though it's only 2 tests, I'd suggest at least looking at whether a table test makes this easier to read.

Reply:
I'm not a fan of table tests and find them less readable. If we won't let the PR through I'll swap them over to table tests, but I'd strongly prefer not to.